main.ml

```diff
@@ -1087,7 +1206,20 @@ let description = String.concat ~sep:" " [
   "the scripts inside.";
 ]
 
+type backend_error = string * (string list) [@@deriving sexp]
+
+let register_exns () =
+  Sexplib.Conv.Exn_converter.add ~finalise:false [%extension_constructor Storage_interface.Backend_error]
+  (function
+    | Storage_interface.Backend_error e -> sexp_of_backend_error e
+    | _ -> assert false);
+  Sexplib.Conv.Exn_converter.add ~finalise:false [%extension_constructor Storage_interface.Backend_error_with_backtrace]
+  (function
+    | Storage_interface.Backend_error_with_backtrace e -> sexp_of_backend_error e
+    | _ -> assert false)
+
 let _ =
+  register_exns ();
```

Comment:
What does this do?

Reply:
It registers pretty printers for the arguments of exceptions, so instead of `(_)` you see the actual argument, for example before you had this:
```
test thread failed with (monitor.ml.Error ("Storage_interface.Backend_error_with_backtrace(_)"))
```

And now you get this instead:
```
test thread failed with (monitor.ml.Error
  (SR_BACKEND_FAILURE
    ("{\"error\":\"UnmarshalException thing=argument missing ty=uri desc=\",\"files\":[\"/usr/lib/python2.7/site-packages/xapi/storage/api/volume.py\",\"/usr/lib/python2.7/site-packages/xapi/storage/api/volume.py\"],\"lines\":[1281,842]}"
      UnmarshalException
      "UnmarshalException thing=argument missing ty=uri desc=")))

```

Reply:
Could we mention this in a comment or change the name of the function to reflect that it registers exception pretty printers?